libraries/Makefile

```diff
@@ -51,28 +51,37 @@ include $(CL_DIR)/hdk.mk
 # flags, and a list of sources.
 include $(LIBRARIES_PATH)/libraries.mk
 
-build: $(LIBRARIES_PATH)/libbsg_manycore_runtime.so.1.0
+_TARGETS :=
+_DOCSTRING :=
 
-install: $(LIBRARIES_PATH)/libbsg_manycore_runtime.so.1.0
-	mv $(notdir $<) /usr/lib64/
-	ln -sf /usr/lib64/$(notdir $<) /usr/lib64/libbsg_manycore_runtime.so.1
-	ln -sf /usr/lib64/$(notdir $<) /usr/lib64/libbsg_manycore_runtime.so
-	cp -t /usr/include $(LIB_HEADERS)
+_DOCSTRING += "Rules from Makefile\n"
+_TARGETS :=
 
+.DEFAULT_GOAL := help
+_TARGETS += "help"
+_DOCSTRING += "    help: \n"
```

Comment:
You know - you could do this kind of thing like this:

```
_DOCSTRING-atargetname := "Help message for a target name"
targets += atargetname
```

then in help you do:
```
help:
    @(foreach t, $(targets), echo "$t:"; echo "    $(_DOCSTRING-$t)";)
```
True story - that works... and then you don't need to report the formatting of your help message everywhere you add a new target.

Not saying that you adopt that strategy here but maybe something to think about in the future.